pilot/pkg/model/service.go

```diff
@@ -844,7 +844,7 @@ func (i AddressInfo) ResourceName() string {
 	case *workloadapi.Address_Workload:
 		ii, _ = netip.AddrFromSlice(addr.Workload.Address)
 	case *workloadapi.Address_Service:
-		ii, _ = netip.AddrFromSlice(addr.Service.Address)
+		ii, _ = netip.AddrFromSlice(addr.Service.Addresses[0].Address)
```

Comment:
Can you add a comment on this method ? What it is is used for, how it deals with multiple addresses in workloads and services ( IP4+IP6, service with multiple addresses, etc). It's very likely to be hard to understand for others too.

I think if we want stable names - we should use the pod hostname, namespace, cluster. And for service too. IP addresses are bad names - multiple values, ephemeral for pods.

Reply:
this has been updated to what is documented in the design doc.  Service keys are `namespace/hostname`